metric/series_grouper.go

```diff
@@ -57,30 +63,48 @@ func (g *SeriesGrouper) Add(
 	return nil
 }
 
+// AddMetric adds a metric to the series, merging with any previous matching metrics.
+func (g *SeriesGrouper) AddMetric(
+	metric telegraf.Metric,
+) {
+	id := groupID(metric.Name(), metric.TagList(), metric.Time())
+	m := g.metrics[id]
+	if m == nil {
+		m = metric.Copy()
+		g.metrics[id] = m
+		g.ordered = append(g.ordered, m)
+	} else {
+		for _, f := range metric.FieldList() {
+			m.AddField(f.Key, f.Value)
+		}
+	}
+}
+
 // Metrics returns the metrics grouped by series and time.
 func (g *SeriesGrouper) Metrics() []telegraf.Metric {
 	return g.ordered
 }
 
-func groupID(measurement string, tags map[string]string, tm time.Time) uint64 {
-	h := fnv.New64a()
-	h.Write([]byte(measurement))
-	h.Write([]byte("\n"))
+var seed = maphash.MakeSeed()
```

Comment:
Can't this be done in `NewSeriesGrouper()` maybe together with the `SetSeed()` call?